### PR TITLE
[WIP] presets - merge default babel configs

### PR DESCRIPTION
**Pull request by igor-dv**
Issue: This PR is based on the #4043 (for the reviewability)

## What I did
I've removed `wrapDefaultBabelConfig` in apps, and added the ability to merge babel configs. This will ease the app creation process and presets usage

### CODE REVIEW

File: lib/core/src/server/config.js [hunk 1]
```diff
@@ -2,16 +2,12 @@ import path from 'path';
...
-
 function customPreset({ configDir }) {
   return serverRequire(path.resolve(configDir, 'presets')) || [];
 }
 
 function getWebpackConfig(options, presets) {
-  const { configDir, wrapDefaultBabelConfig = noopWrapper } = options;
-
-  const babelOptions = presets.extendBabel({}, { configDir, wrapDefaultBabelConfig });
+  const babelOptions = presets.extendBabel({}, options);
```
Comment (Hypnosphi) [inline lib/core/src/server/config.js]: This breaks Babel 6 support, because it adds our defaults (which are Babel 7 only) to all configs

Comment (Hypnosphi) [reply lib/core/src/server/config.js]: We really need separate `extendDefaultBabel` and `extendBabel`

Comment (Hypnosphi) [reply lib/core/src/server/config.js]: Actually, react-docgen is probably the only usecase for the latter

Comment (igor-dv) [reply lib/core/src/server/config.js]: You mean `react-docgen` is the only working usecase?

Comment (Hypnosphi) [reply lib/core/src/server/config.js]: It's the only plugin that we need to add to any configuration because it typically isn't used in user app

**Pull request merged**